app/controllers/straws_controller.rb

```diff
@@ -2,16 +2,59 @@ class StrawsController < ApplicationController
   skip_before_action :verify_authenticity_token, only: :create
 
   def create
+# basic parsing
     straw_usernames = params[:text].split(' ')
-    short_straw_username = straw_usernames.sample
     host_username = params[:user_name]
 
-    straw_response = "@#{host_username} "
-    straw_response += "created a straw poll for #{straw_usernames.to_sentence}.\n\n"
-    straw_response += "*#{short_straw_username} has drawn the short straw*. Sucka!"
+# advanced parsing
+    straw_task_description = params[:text].scan(/\"(.+)\"/).flatten.join
+    straw_selected_users = params[:text].scan(/@[a-zA-Z0-9_\-\.]{1,21}\b/)
+
+# selection logic
+    short_straw_username = straw_selected_users.sample
+    straw_list = ""
+    straw_selected_users.each { |username| straw_list += "#{username}\n" }
+
+# message creation
+    straw_host_message = "A straw drawing was created by @#{host_username} "
+    straw_selection_message = "*<#{short_straw_username}> has drawn the short straw!*"
 
     slack_response = {
-      "text": straw_response,
+      "attachments": [
+          {
+              "fallback": "Required plain-text summary of the attachment.",
+              "pretext": straw_host_message,
+              "color": "#70CADB",
+              "title": "Let's draw straws for it!",
+              "title_link": "https://trello.com/b/4TTrKn3s/draw-straws",
+              "text": straw_task_description,
+              "fields": [
+                  {
+                      "title": "drawing from:",
+                      "value": straw_list,
+                      "short": false
+                  }
+              ],
+              "mrkdwn_in": ["text"]
+          },
+          {
+              "fallback": "Required plain-text summary of the attachment.",
+              "color": "#FFA500",
+              "text": straw_selection_message,
+              "mrkdwn_in": ["text"]
+          },
+          {
+              "fallback": "Required plain-text summary of the attachment.",
+              "color": "#70CADB",
+              "text": "_Thanks for drawing! Questions? Visit our <http://letsdrawstraws.com|help center>._",
+              "image_url": "http://my-website.com/path/to/image.jpg",
+              "thumb_url": "http://example.com/path/to/thumb.png",
+              "footer": "DrawStraws",
+              "footer_icon": "https://platform.slack-edge.com/img/default_application_icon.png",
```

Comment:
I'm fine with leaving these to a later PR / Trello card, but can we just cut them for now?